### ISSUE TITLE
Imported Guest replays point to osu account with name Guest.

### ISSUE THREAD

**Issue by Vergenter** [ppy/osu#20496]
### Type

Game behaviour

### Bug description

NOTE: There is no problem if I use export on normal local guest replay, remove it and readd from export.
I wanted to import replays from other osu!lazer instalation.
I used python script to identify which files in  ~/.local/share/osu/files folder are replays.(old drive)
```py
def is_file_replay(filepath: str) -> bool:
    REPLAY_HEADER = b"\x00\x80\xc3\xc9\x01\x0b"
    with open(filepath, "rb") as f:
        firstline = f.readline()
        return firstline.startswith(REPLAY_HEADER)
```
I copied them to different folder and added extension `.osr`
After import all guest replays point to player [Guest](https://osu.ppy.sh/users/11666366).
As you can see on the screenshot:
![osu_2022-09-27_02-24-26](https://user-images.githubusercontent.com/37329111/192404352-e4ba984f-3d75-4d52-978e-35b9ecf5d38e.jpg)
I attached `.osr` file from screenshot. 

### Screenshots or videos

[GUEST_REPLAY.zip](https://github.com/ppy/osu/files/9650472/GUEST_REPLAY.zip)
![osu_2022-09-27_02-24-26](https://user-images.githubusercontent.com/37329111/192404352-e4ba984f-3d75-4d52-978e-35b9ecf5d38e.jpg)

### Version

2022.923.1-lazer

### Logs

[database.log](https://github.com/ppy/osu/files/9650547/database.log)
[network.log](https://github.com/ppy/osu/files/9650548/network.log)
[performance.log](https://github.com/ppy/osu/files/9650549/performance.log)
[runtime.log](https://github.com/ppy/osu/files/9650550/runtime.log)

**Comment by peppy** [comment]
I don't think this is a flow we can support. Can you better describe your use case?
